dotcms-integration/src/test/java/com/dotmarketing/factories/MultiTreeAPITest.java

```diff
@@ -2913,4 +2903,1249 @@ public void deletePersonalizationSpecificVariant() throws DotDataException {
+                new String(new char[pageIds.size() - 1]).replace("\0", ", ?") + ") order by parent1";
+
+        final DotConnect dotConnect = new DotConnect().setSQL(getMultiTreeQuery);
+
+        pageIds.stream().forEach(pageId -> dotConnect.addParam(pageId));
+        final ArrayList<Map<String, Object>> multiTreeResult = dotConnect.loadResults();
+        int multiTreeCurrentIndex = 0;
+
+        for (int i = 0; i < pagesNumber; i++) {
+
+            final Map<String, List<Map<String, Object>>> groupedByInstanceId = getMultiTreeByPageOrderByInstanceId(pageIds.get(i),
+                    multiTreeResult, multiTreeCurrentIndex);
+
+            final int size = groupedByInstanceId.values().stream().map(List::size).reduce(0, Integer::sum);
+            multiTreeCurrentIndex = multiTreeCurrentIndex + size;
+
+            assertEquals(5, size);
+
+            for (final String intanceId : groupedByInstanceId.keySet()) {
+                final List<Map<String, Object>> multiTrees = groupedByInstanceId.get(intanceId);
+
+                switch (intanceId) {
+                    case "1":
+                        assertEquals(1, multiTrees.size());
+                        assertEquals(contentlet_4.getIdentifier(), multiTrees.get(0).get("child"));
+                        break;
+                    case "2":
+                        assertEquals(1, multiTrees.size());
+                        assertEquals(contentlet_1.getIdentifier(), multiTrees.get(0).get("child"));
+                        break;
+                    case "3":
+                        assertEquals(1, multiTrees.size());
+                        assertEquals(contentlet_2.getIdentifier(), multiTrees.get(0).get("child"));
+                        break;
+                    case "4":
+                        assertEquals(2, multiTrees.size());
+
+                        List<String> contentlets = multiTrees.stream().map(multiTree -> multiTree.get("child").toString())
+                                .collect(Collectors.toList());
+                        assertTrue(contentlets.contains(contentlet_3.getIdentifier()));
+                        assertTrue(contentlets.contains(contentlet_5.getIdentifier()));
+                        break;
+                    default:
+                        throw new AssertionError("UUID not expected: " + intanceId);
+                }
+            }
+        }
+    }
+
+    /**
+     * Method to test: {@link MultiTreeAPIImpl#updateMultiTrees(String, LayoutChanges)}
+     * When: You have 100 Page with 4 containers all of them are different instances of the same containers,
+     * and you have 5 Contentlets add as follows:
+     * - Contentlet_1 : Add to the instance 1
+     * - Contentlet_2 : Add to the instance 2
+     * - Contentlet_3 : Add to the instance 3
+     * - Contentlet_5 : Add to the instance 3
+     * - Contentlet_4 : Add to the instance 4
+     *
+     * Should: Return on list of 100 millis
```

Comment:
Return the list of pages under 100 millis*

Reply:
done https://github.com/dotCMS/core/pull/28128/commits/207aa29740f2ac300f87685154ada6b9af84c595